Src/library/net/network/socket/network_socket.e

```diff
@@ -116,76 +116,92 @@ feature -- Status report
 			-- Is reuse_address option set?
 		require
 			socket_exists: exists
-		local
-			reuse: INTEGER
 		do
-			reuse := c_get_sock_opt_int (descriptor, level_sol_socket, so_reuse_addr);
-			Result := reuse /= 0
+			Result := c_get_sock_opt_int (descriptor, level_sol_socket, so_reuse_addr) /= 0
 		end
 
 	is_valid_peer_address (addr: attached separate like address): BOOLEAN
 		do
-			Result := (addr.family = af_inet or else addr.family = af_inet6)
+			Result := addr.family = af_inet or else addr.family = af_inet6
 		end
 
 	is_valid_family (addr: attached like address): BOOLEAN
 		do
-			Result := (addr.family = af_inet or else addr.family = af_inet6)
+			Result := addr.family = af_inet or else addr.family = af_inet6
 		end
 
 	ready_for_reading: BOOLEAN
 			-- Is data available for reading from the socket within
 			-- `timeout' seconds?
 		require
 			socket_exists: exists
-		local
-			retval: INTEGER
 		do
-			retval := c_select_poll_with_timeout (descriptor, True, timeout)
-			Result := (retval > 0)
+			Result := c_select_poll_with_timeout (descriptor, True, timeout_ns) > 0
 		end
 
 	ready_for_writing: BOOLEAN
 			-- Can data be written to the socket within `timeout' seconds?
 		require
 			socket_exists: exists
-		local
-			retval: INTEGER
 		do
-			retval := c_select_poll_with_timeout (descriptor, False, timeout)
-			Result := (retval > 0)
+			Result := c_select_poll_with_timeout (descriptor, False, timeout_ns) > 0
 		end
 
 	has_exception_state: BOOLEAN
 			-- Is socket in exception state within `timeout' seconds?
 		require
 			socket_exists: exists
-		local
-			retval: INTEGER
 		do
-			retval := c_check_exception_with_timeout (descriptor, timeout)
-			Result := (retval > 0)
+			Result := c_check_exception_with_timeout (descriptor, timeout_ns) > 0
 		end
 
 	timeout: INTEGER
 			-- Duration of timeout in seconds
+		obsolete
+			"Use `timeout_ns` using nanoseconds [2018-01-15]"
+		do
+			Result := nanoseconds_to_seconds (timeout_ns)
+			if Result = 0 then
+					-- 0 may have different meaning, so return the lowest possible value.
```

Comment:
I don't think we should be doing that but I don't know why you added this in the first place.

Reply:
I agree, but this is current behavior ... I search and find nothing that would explain why we should not accept 0 as timeout.
However here we should first check if timeout_ns is not zero.